libdnf/hy-package.cpp

```diff
@@ -943,6 +945,39 @@ dnf_package_get_files(DnfPackage *pkg)
     return (gchar**)g_ptr_array_free (ret, FALSE);
 }
 
+/**
+ * dnf_package_get_changelogs:
+ * @pkg: a #DnfPackage instance.
+ *
+ * Gets the list of changelogs for the package.
+ *
+ * Returns: (transfer container) (element-type Changelog): a list
+ *
+ * Since: 0.19.1
+ */
+std::vector<libdnf::Changelog>
+dnf_package_get_changelogs(DnfPackage *pkg)
+{
+    DnfPackagePrivate *priv = GET_PRIVATE(pkg);
+    Pool *pool = dnf_package_get_pool(pkg);
+    Solvable *s = get_solvable(pkg);
+    Dataiterator di;
+    std::vector<libdnf::Changelog> changelogslist;
+
+    dataiterator_init(&di, pool, s->repo, priv->id, SOLVABLE_CHANGELOG_AUTHOR, NULL, 0);
+    dataiterator_prepend_keyname(&di, SOLVABLE_CHANGELOG);
+    while (dataiterator_step(&di)) {
+        dataiterator_setpos_parent(&di);
+        changelogslist.push_back(libdnf::Changelog{
```

Comment:
Please use `emplace_back` (constructs an element in-place at the end) instead of creating it and adding (moving) to the end using `push_back`.
```
changelogslist.emplace_back(
    static_cast<time_t>(pool_lookup_num(pool, SOLVID_POS, SOLVABLE_CHANGELOG_TIME, 0)),
    pool_lookup_str(pool, SOLVID_POS, SOLVABLE_CHANGELOG_AUTHOR),
    pool_lookup_str(pool, SOLVID_POS, SOLVABLE_CHANGELOG_TEXT));
```